content_harvester/by_record.py

```diff
@@ -221,7 +221,15 @@ def check_component_cache(
 
             # Do a head request to get the current ETag and
             # Last-Modified values, used to create a cache key
-            head_resp = http_session.head(**asdict(request))
+            if 'nuxeo' in request.url:
+                # The S3 presigned URLs from Nuxeo are good for GET requests only
+                # so do a GET request that mimics a head request
+                head_resp = http_session.get(
+                    **asdict(request),
+                    headers={"Range": "bytes=0-0"}
+                )
+            else:
+                head_resp = http_session.head(**asdict(request), allow_redirects=True)
```

Comment:
requests doc indicate that is `True` by default 

> allow_redirects ([bool](https://docs.python.org/3/library/functions.html#bool)) – (optional) Boolean. Enable/disable GET/OPTIONS/POST/PUT/PATCH/DELETE/HEAD redirection. Defaults to True.

Why wasn't that working before?

Reply:
In the docs I’m looking at (maybe not the version we’re on, but) it says “ By default Requests will perform location redirection for all verbs except HEAD.”

https://docs.python-requests.org/en/latest/user/quickstart/#redirection-and-history

Reply:
Lol, it's the same site - https://docs.python-requests.org/en/latest/api/#module-requests